include/work.h

```diff
@@ -7,6 +7,12 @@
 #include "util.h"
 #include "logger.h"
 
+/*
+ * (Defined in lib/work.c)
+ * If this is greater than 0, the number of threads grows based on it.
+ */
+extern size_t max_dynamic_threads;
```

Comment:
Do we need to export this variable? Providing a function that just initializes `max_dynamic_threads` seems to be cleaner for separation because the variable doesn't need to be read from sheep side.